### ISSUE TITLE
use re 'eval'; lost in (??{"(?{1})"})

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#6717]
Migrated from [rt.perl.org#23569](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=23569) (status was 'resolved')

Searchable as RT23569$

**Comment by p5pRT** [comment]
### From bsb@strategicdata.com.au

#### Created by bsb@strategicdata.com.au



This code generates an "Eval\-group \.\.\." error despite the  presense
of "use re qw\(eval\)"\.

$ perl \-Mre=eval \-e '/ \(??\{ "\(?\{1\}\)" \}\) $\_ /x'
Eval\-group not allowed at runtime\, use re 'eval' in regex m/\(?\{1\}\)/ at \-e line 1\.

It seems like the 'eval' hint doesn't reach the right scope at the right phase\.

Returning a qr//'ed object is a workaround\.

$ perl  \-e '/ \(?\{1\}\) $\_ /x'
Eval\-group not allowed at runtime\, use re 'eval' in regex m/ \(?\{1\}\)  / at \-e line 1\.

$ perl \-Mre=eval \-e '/ \(?\{1\}\) $\_ /x'

\# This is the odd one
$ perl \-Mre=eval \-e '/ \(??\{ "\(?\{1\}\)" \}\) $\_ /x'
Eval\-group not allowed at runtime\, use re 'eval' in regex m/\(?\{1\}\)/ at \-e line 1\.

$ perl \-Mre=eval \-e '/ \(??\{ qr"\(?\{1\}\)" \}\) $\_ /x'

<details><summary>Perl Info</summary>

```

Flags:
    category=core
    severity=low

Site configuration information for perl v5.6.1:

Configured by buildd at Sat Nov 30 17:45:17 UTC 2002.

Summary of my perl5 (revision 5.0 version 6 subversion 1) configuration:
  Platform:
    osname=linux, osvers=2.4.19, archname=i386-linux
    uname='linux cyberhq 2.4.19 #1 smp sun aug 4 11:30:45 pdt 2002 i686 unknown '
    config_args='-Dccflags=-DDEBIAN -Dcccdlflags=-fPIC -Darchname=i386-linux -Dprefix=/usr -Dprivlib=/usr/share/perl/5.6.1 -Darchlib=/usr/lib/perl/5.6.1 -Dvendorprefix=/usr -Dvendorlib=/usr/share/perl5 -Dvendorarch=/usr/lib/perl5 -Dsiteprefix=/usr/local -Dsitelib=/usr/local/share/perl/5.6.1 -Dsitearch=/usr/local/lib/perl/5.6.1 -Dman1dir=/usr/share/man/man1 -Dman3dir=/usr/share/man/man3 -Dman1ext=1 -Dman3ext=3perl -Dpager=/usr/bin/sensible-pager -Uafs -Ud_csh -Uusesfio -Duseshrplib -Dlibperl=libperl.so.5.6.1 -Dd_dosuid -des'
    hint=recommended, useposix=true, d_sigaction=define
    usethreads=undef use5005threads=undef useithreads=undef usemultiplicity=undef
    useperlio=undef d_sfio=undef uselargefiles=define usesocks=undef
    use64bitint=undef use64bitall=undef uselongdouble=undef
  Compiler:
    cc='cc', ccflags ='-DDEBIAN -fno-strict-aliasing -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64',
    optimize='-O2',
    cppflags='-DDEBIAN -fno-strict-aliasing -I/usr/local/include'
    ccversion='', gccversion='2.95.4 20011002 (Debian prerelease)', gccosandvers=''
    intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=1234
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
    ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=8
    alignbytes=4, usemymalloc=n, prototype=define
  Linker and Libraries:
    ld='cc', ldflags =' -L/usr/local/lib'
    libpth=/usr/local/lib /lib /usr/lib
    libs=-lgdbm -ldb -ldl -lm -lc -lcrypt
    perllibs=-ldl -lm -lc -lcrypt
    libc=/lib/libc-2.2.5.so, so=so, useshrplib=true, libperl=libperl.so.5.6.1
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-rdynamic'
    cccdlflags='-fPIC', lddlflags='-shared -L/usr/local/lib'

Locally applied patches:
    


@INC for perl v5.6.1:
    /home/bsb/japh
    /usr/local/lib/perl/5.6.1
    /usr/local/share/perl/5.6.1
    /usr/lib/perl5
    /usr/share/perl5
    /usr/lib/perl/5.6.1
    /usr/share/perl/5.6.1
    /usr/local/lib/site_perl
    .


Environment for perl v5.6.1:
    HOME=/home/bsb
    LANG=en_AU
    LANGUAGE (unset)
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/home/bsb/bin:/usr/local/bin:/usr/bin:/bin:/usr/bin/X11:/usr/games:~/bin:~/Komodo-2.0
    PERL5LIB=/home/bsb/japh
    PERL_BADLANG (unset)
    SHELL=/bin/bash



```
</details>

**Comment by p5pRT** [comment]
### From @rgarcia

Brad Bowman \(via RT\) wrote&#8203;:
> This code generates an "Eval\-group \.\.\." error despite the  presense
> of "use re qw\(eval\)"\.
> 
> $ perl \-Mre=eval \-e '/ \(??\{ "\(?\{1\}\)" \}\) $\_ /x'
> Eval\-group not allowed at runtime\, use re 'eval' in regex m/\(?\{1\}\)/ at \-e line 1\.

This is apparently just another instance of the compiler hints not being
propagated to eval\-strings \(except "integer"\, "strict refs" and a few
others\)\. This can't be fixed non\-kludgily without an overhaul of the hint
system\.

**Comment by p5pRT** [comment]
### From @cpansprout

This has been fixed by the commits leading up to eb58a7e122\.

**Comment by p5pRT** [comment]
### From [Unknown Contact. See original ticket]

This has been fixed by the commits leading up to eb58a7e122\.

**Comment by p5pRT** [comment]
@cpansprout - Status changed from 'open' to 'resolved'
